src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -581,6 +582,22 @@ impl CatalogState {
                                         diff,
                                     )
                                 }
+                                "sql-server" => {
+                                    mz_ore::soft_assert_eq_no_log!(external_reference.len(), 3);
+                                    // The left-most qualification of SQL Server tables is
+                                    // the database, but this information is redundant
+                                    // because each SQL Server connection connects to
+                                    // only one database.
+                                    let schema_name = external_reference[1].to_ast_string_simple();
+                                    let table_name = external_reference[2].to_ast_string_simple();
```

Comment:
I think this will panic in prod if external_reference.len() <3. soft_assert_eq_no_log won't bail from this branch in prod. Might be worth switching to an explicit assert so we can at least add a message to the panic rather than an out-of-bounds